### PR TITLE
Alpaca

### PR DESCRIPTION
**Pull request by shershah010**
### Not really tested

Note:
* Alpaca does not support options
* Free users cannot access crypto

Questions:
* Alpaca supports `stop` orders, how to  we handle those?
* If a buy or sell fails during a algo's run, should the entire process be terminated to prevent potential losses?

### CODE REVIEW

**Comment by tfukaza** [comment]
- We should NOT support stop orders, IMO the user should implement stop orders using the algorithm.
- That's some thing we'll have to discuss, possibly in another ticket.

File: harvest/api/alpaca.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+            else:
+                self.watch_stock.append(s)
+
+        self.option_cache = {}
+        super().setup(watch, interval, interval, trader, trader_main)
+
+    def exit(self):
+        self.option_cache = {}
+
+    def main(self):
```
Comment (tfukaza) [inline harvest/api/alpaca.py:43]: Alpaca offers a streaming API, so it is better to use that instead of polling.

**Review by tfukaza — CHANGES_REQUESTED** [review]
Looks great! Just need to use the streaming API for Alpaca.

File: harvest/api/alpaca.py (hunk 1 above)
Comment (shershah010) [reply harvest/api/alpaca.py:84]: Now uses the streaming service to get the latest price. Note that I think it updates per minute so if someone sets the interval to 1HR they will still get the price per minute and not aggregated over an hour.

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
     - name: Run tests
       run: |
         pip install coverage
-        coverage run --source harvest -m unittest discover -s tests
+        coverage run --source harvest --omit harvest/api/robinhood.py,harvest/api/alpaca.py -m unittest discover -s tests
```
Comment (shershah010) [inline .github/workflows/run-tests.yml:44]: ignore files that have no tests in the coverage

File: harvest/api/alpaca.py [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+        elif f'@{symbol}' in self.watch_crypto:
+            self.data['cryptos'][f'@{symbol}'] = self._format_df(df, f'@{symbol}')
+        self.data_lock.release()
+
+
+    def no_secret(self, path: str) -> bool:
+        return self.create_secret(path)
+
+    def capture_data(self):
+        loop = asyncio.new_event_loop()
```
Comment (shershah010) [inline harvest/api/alpaca.py:59]: necessary since the thread has no event loop.

**Review by tfukaza — APPROVED** [review]

**Pull request merged**